From 5830d38a3c1936c5b38d563b4cb11c9ae414bf32 Mon Sep 17 00:00:00 2001
From: Stas Boukarev <stassats@gmail.com>
Date: Wed, 28 May 2014 18:19:36 +0400
Subject: [PATCH 011/100] Enable GENCGC on ARM.

---
 NEWS           | 1 +
 make-config.sh | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/make-config.sh b/make-config.sh
index c12c2d2..21639fd 100755
--- a/make-config.sh
+++ b/make-config.sh
@@ -675,7 +675,7 @@ elif [ "$sbcl_arch" = "hppa" ]; then
     printf ' :stack-allocatable-vectors :stack-allocatable-fixed-objects' >> $ltf
     printf ' :stack-allocatable-lists' >> $ltf
 elif [ "$sbcl_arch" = "arm" ]; then
-    printf ' :cheneygc :linkage-table :alien-callbacks' >> $ltf
+    printf ' :gencgc :linkage-table :alien-callbacks' >> $ltf
     # As opposed to soft-float or FPA, we support VFP only (and
     # possibly VFPv2 and higher only), but we'll leave the obvious
     # hooks in for someone to add the support later.
-- 
1.9.3

