diff -up sbcl-1.0.15/src/runtime/x86-assem.S.binutils sbcl-1.0.15/src/runtime/x86-assem.S
--- sbcl-1.0.15/src/runtime/x86-assem.S.binutils	2007-09-26 06:24:02.000000000 -0500
+++ sbcl-1.0.15/src/runtime/x86-assem.S	2008-04-10 12:52:20.000000000 -0500
@@ -130,7 +130,7 @@ GNAME(call_into_c):
 
 /* Check for a return FP value. */
 	fxam
-	fnstsw	%eax
+	fnstsw	%ax
 	andl	$0x4500,%eax
 	cmpl	$0x4100,%eax
 	jne	Lfp_rtn_value
