diff -up sbcl-1.0.51/generate-version.sh.gitfix sbcl-1.0.51/generate-version.sh
--- sbcl-1.0.51/generate-version.sh.gitfix	2011-08-20 18:15:12.000000000 -0500
+++ sbcl-1.0.51/generate-version.sh	2011-08-22 08:17:12.575478114 -0500
@@ -2,7 +2,7 @@
 # Not a shell script, but something intended to be sourced from shell scripts
 git_available_p() {
     # Check that (1) we have git (2) this is a git tree.
-    if ( which git >/dev/null 2>/dev/null && git describe >/dev/null 2>/dev/null )
+    if ( test -d .git && which git >/dev/null 2>/dev/null && git describe >/dev/null 2>/dev/null )
     then
         echo "ok"
     else
