diff -up sbcl-1.2.8/contrib/asdf-module.mk.optflags sbcl-1.2.8/contrib/asdf-module.mk
--- sbcl-1.2.8/contrib/asdf-module.mk.optflags	2015-01-31 08:24:55.000000000 -0600
+++ sbcl-1.2.8/contrib/asdf-module.mk	2015-02-13 12:53:48.144358286 -0600
@@ -27,6 +27,8 @@ ifeq (Linux,$(UNAME))
   EXTRA_CFLAGS=-D_GNU_SOURCE
 endif
 
+EXTRA_CFLAGS += $(RPM_OPT_FLAGS)
+
 export CC SBCL EXTRA_CFLAGS EXTRA_LDFLAGS
 
 all: $(FASL) $(ASD) $(EXTRA_ALL_TARGETS)
diff -up sbcl-1.2.8/src/runtime/Config.arm-linux.optflags sbcl-1.2.8/src/runtime/Config.arm-linux
--- sbcl-1.2.8/src/runtime/Config.arm-linux.optflags	2015-01-31 08:24:57.000000000 -0600
+++ sbcl-1.2.8/src/runtime/Config.arm-linux	2015-02-13 12:53:48.144358286 -0600
@@ -9,7 +9,7 @@
 # provided with absolutely no warranty. See the COPYING and CREDITS
 # files for more information.
 
-CFLAGS = -g -marm -march=armv5
+CFLAGS += $(RPM_OPT_FLAGS)
 #LINKFLAGS += -v -rdynamic -m32
 NM = ./linux-nm
 
diff -up sbcl-1.2.8/src/runtime/Config.ppc-linux.optflags sbcl-1.2.8/src/runtime/Config.ppc-linux
--- sbcl-1.2.8/src/runtime/Config.ppc-linux.optflags	2015-01-31 08:24:57.000000000 -0600
+++ sbcl-1.2.8/src/runtime/Config.ppc-linux	2015-02-13 12:53:48.144358286 -0600
@@ -9,7 +9,7 @@
 # provided with absolutely no warranty. See the COPYING and CREDITS
 # files for more information.
 
-CFLAGS = -g -m32
+CFLAGS += -g -m32
 LINKFLAGS += -v -rdynamic -m32
 NM = ./linux-nm
 
@@ -19,6 +19,8 @@ ARCH_SRC = ppc-arch.c
 OS_SRC = linux-os.c ppc-linux-os.c
 OS_LIBS = -ldl
 
+CFLAGS += $(RPM_OPT_FLAGS)
+
 ifdef LISP_FEATURE_GENCGC
   GC_SRC = gencgc.c
 else
diff -up sbcl-1.2.8/src/runtime/Config.sparc-linux.optflags sbcl-1.2.8/src/runtime/Config.sparc-linux
--- sbcl-1.2.8/src/runtime/Config.sparc-linux.optflags	2015-01-31 08:24:57.000000000 -0600
+++ sbcl-1.2.8/src/runtime/Config.sparc-linux	2015-02-13 12:53:48.145358308 -0600
@@ -28,6 +28,8 @@ else
   GC_SRC = cheneygc.c
 endif
 
+CFLAGS += $(RPM_OPT_FLAGS)
+
 # Nothing to do for after-grovel-headers.
 .PHONY: after-grovel-headers
 after-grovel-headers:
diff -up sbcl-1.2.8/src/runtime/Config.x86-64-linux.optflags sbcl-1.2.8/src/runtime/Config.x86-64-linux
--- sbcl-1.2.8/src/runtime/Config.x86-64-linux.optflags	2015-02-13 12:53:51.220426514 -0600
+++ sbcl-1.2.8/src/runtime/Config.x86-64-linux	2015-02-13 12:54:35.302379698 -0600
@@ -44,6 +44,8 @@ endif
 
 CFLAGS += -fno-omit-frame-pointer
 
+CFLAGS += $(RPM_OPT_FLAGS)
+
 GC_SRC = gencgc.c
 
 # Hardened GCC enables PIE by default
diff -up sbcl-1.2.8/src/runtime/Config.x86-linux.optflags sbcl-1.2.8/src/runtime/Config.x86-linux
--- sbcl-1.2.8/src/runtime/Config.x86-linux.optflags	2015-01-31 08:24:57.000000000 -0600
+++ sbcl-1.2.8/src/runtime/Config.x86-linux	2015-02-13 12:53:51.220426514 -0600
@@ -49,6 +49,8 @@ LINKFLAGS += -nopie
 LDFLAGS += -nopie
 endif
 
+CFLAGS += $(RPM_OPT_FLAGS)
+
 GC_SRC = gencgc.c
 
 # Nothing to do for after-grovel-headers.
