diff -up sbcl-1.4.14/src/runtime/linux-os.c.personality sbcl-1.4.14/src/runtime/linux-os.c
--- sbcl-1.4.14/src/runtime/linux-os.c.personality	2018-12-24 08:28:30.346162495 -0600
+++ sbcl-1.4.14/src/runtime/linux-os.c	2018-12-24 08:31:24.922224890 -0600
@@ -323,6 +323,10 @@ int os_preinit(char *argv[], char *envp[
                     setenv("SBCL_IS_RESTARTING", "T", 1);
                     runtime[i] = '\0';
                     execv(runtime, argv);
+                } else {
+                    /* if /proc isn't available (like in chroot builds, like mock), 
+                     * try using execvp with argv[0] instead */
+                    execvp(argv[0], argv);
                 }
             }
             /* Either changing the personality or execve() failed.
