--- sbcl-1.0.2/make-target-contrib.sh.permissive	2007-01-18 09:12:03.000000000 -0600
+++ sbcl-1.0.2/make-target-contrib.sh	2007-01-25 11:22:14.000000000 -0600
@@ -54,7 +54,7 @@
 for i in contrib/*; do
     test -d $i && test -f $i/Makefile || continue;
     # export INSTALL_DIR=$SBCL_HOME/`basename $i `
-    test -f $i/test-passed && rm $i/test-passed
+    #test -f $i/test-passed && rm $i/test-passed
     # hack to get exit codes right.
     if $GNUMAKE -C $i test 2>&1 && touch $i/test-passed ; then
 	:
