diff -up sbcl-2.0.1/make.sh.me sbcl-2.0.1/make.sh
--- sbcl-2.0.1/make.sh.me	2020-02-24 10:57:12.072885605 +0100
+++ sbcl-2.0.1/make.sh	2020-02-24 11:03:04.997282509 +0100
@@ -24,7 +24,7 @@ export LANG LC_ALL
 # thing" when run on the target machine, with the minor caveat that
 # any --xc-host parameter should be suitable for the host machine
 # instead of the target.
-sh make-config.sh "$@" || exit $?
+sh -x make-config.sh "$@" || exit $?
 
 . output/prefix.def
 . output/build-config
@@ -76,11 +76,11 @@ maybetime() {
         $@
     fi
 }
-maybetime sh make-host-1.sh
-maybetime sh make-target-1.sh
-maybetime sh make-host-2.sh
-maybetime sh make-target-2.sh
-maybetime sh make-target-contrib.sh
+maybetime sh -x make-host-1.sh
+maybetime sh -x make-target-1.sh
+maybetime sh -x make-host-2.sh
+maybetime sh -x make-target-2.sh
+maybetime sh -x make-target-contrib.sh
 
 # contrib/Makefile shouldn't be counted in NCONTRIBS.
 # "find contrib/* -name Makefile" would still find contrib/./Makefile.
