diff -up sbcl-1.3.2/contrib/asdf-module.mk.optflags sbcl-1.3.2/contrib/asdf-module.mk
--- sbcl-1.3.2/contrib/asdf-module.mk.optflags	2016-01-31 10:14:20.000000000 -0600
+++ sbcl-1.3.2/contrib/asdf-module.mk	2016-02-01 07:03:48.705047251 -0600
@@ -24,6 +24,8 @@ ifeq (Linux,$(UNAME))
   EXTRA_CFLAGS=-D_GNU_SOURCE
 endif
 
+EXTRA_CFLAGS += $(RPM_OPT_FLAGS)
+
 export CC SBCL EXTRA_CFLAGS
 
 all: $(FASL) $(ASD)
diff -up sbcl-1.3.2/src/runtime/Config.arm-linux.optflags sbcl-1.3.2/src/runtime/Config.arm-linux
--- sbcl-1.3.2/src/runtime/Config.arm-linux.optflags	2016-02-01 07:03:48.705047251 -0600
+++ sbcl-1.3.2/src/runtime/Config.arm-linux	2016-02-01 07:04:44.046099199 -0600
@@ -9,7 +9,7 @@
 # provided with absolutely no warranty. See the COPYING and CREDITS
 # files for more information.
 
-CFLAGS += -marm -march=armv5
+CFLAGS += $(RPM_OPT_FLAGS)
 NM = ./linux-nm
 
 ASSEM_SRC = arm-assem.S ldso-stubs.S
diff -up sbcl-1.3.2/src/runtime/Config.ppc-linux.optflags sbcl-1.3.2/src/runtime/Config.ppc-linux
--- sbcl-1.3.2/src/runtime/Config.ppc-linux.optflags	2016-01-31 10:14:21.000000000 -0600
+++ sbcl-1.3.2/src/runtime/Config.ppc-linux	2016-02-01 07:03:48.706047270 -0600
@@ -19,6 +19,8 @@ ARCH_SRC = ppc-arch.c
 OS_SRC = linux-os.c ppc-linux-os.c
 OS_LIBS = -ldl
 
+CFLAGS += $(RPM_OPT_FLAGS)
+
 ifdef LISP_FEATURE_GENCGC
   GC_SRC = gencgc.c
 else
diff -up sbcl-1.3.2/src/runtime/Config.sparc-linux.optflags sbcl-1.3.2/src/runtime/Config.sparc-linux
--- sbcl-1.3.2/src/runtime/Config.sparc-linux.optflags	2016-01-31 10:14:21.000000000 -0600
+++ sbcl-1.3.2/src/runtime/Config.sparc-linux	2016-02-01 07:03:48.706047270 -0600
@@ -28,6 +28,8 @@ else
   GC_SRC = cheneygc.c
 endif
 
+CFLAGS += $(RPM_OPT_FLAGS)
+
 # Nothing to do for after-grovel-headers.
 .PHONY: after-grovel-headers
 after-grovel-headers:
diff -up sbcl-1.3.2/src/runtime/Config.x86-64-linux.optflags sbcl-1.3.2/src/runtime/Config.x86-64-linux
--- sbcl-1.3.2/src/runtime/Config.x86-64-linux.optflags	2016-01-31 10:14:21.000000000 -0600
+++ sbcl-1.3.2/src/runtime/Config.x86-64-linux	2016-02-01 07:03:48.707047289 -0600
@@ -44,6 +44,8 @@ endif
 
 CFLAGS += -fno-omit-frame-pointer
 
+CFLAGS += $(RPM_OPT_FLAGS)
+
 GC_SRC = gencgc.c
 
 # Hardened GCC enables PIE by default
diff -up sbcl-1.3.2/src/runtime/Config.x86-linux.optflags sbcl-1.3.2/src/runtime/Config.x86-linux
--- sbcl-1.3.2/src/runtime/Config.x86-linux.optflags	2016-01-31 10:14:21.000000000 -0600
+++ sbcl-1.3.2/src/runtime/Config.x86-linux	2016-02-01 07:03:48.707047289 -0600
@@ -49,6 +49,8 @@ LINKFLAGS += -nopie
 LDFLAGS += -nopie
 endif
 
+CFLAGS += $(RPM_OPT_FLAGS)
+
 GC_SRC = gencgc.c
 
 # Nothing to do for after-grovel-headers.
