diff -up sbcl-1.3.16/tools-for-build/Makefile.format_security sbcl-1.3.16/tools-for-build/Makefile
--- sbcl-1.3.16/tools-for-build/Makefile.format_security	2017-03-27 04:57:35.000000000 -0500
+++ sbcl-1.3.16/tools-for-build/Makefile	2017-03-27 10:40:19.925856854 -0500
@@ -17,7 +17,7 @@ LDLIBS:=$(OS_LIBS)
 all: grovel-headers determine-endianness where-is-mcontext \
         modify-ldt-struct-name sigaction-sa-nodefer-works-test
 
-grovel-headers: CPPFLAGS+=-Wno-format
+grovel-headers: CPPFLAGS+=-Wno-format -Wno-format-security
 
 clean:
 	rm -f *.o grovel-headers determine-endianness where-is-mcontext \
