--- sbcl-0.9.5/make-config.sh.make-config-fix	2005-09-21 10:00:33.000000000 -0500
+++ sbcl-0.9.5/make-config.sh	2005-09-28 11:42:29.135788304 -0500
@@ -208,7 +208,7 @@
     # cross-compilers!
     #
     # FIXME: integrate to grovel-features, mayhaps
-    $GNUMAKE -C tools-for-build determine-endianness -I src/runtime
+    $GNUMAKE -C tools-for-build determine-endianness -I ../src/runtime
     tools-for-build/determine-endianness >> $ltf
 elif [ "$sbcl_arch" = "ppc" -a "$sbcl_os" = "linux" ]; then
     # Use a C program to detect which kind of glibc we're building on,
@@ -217,8 +217,8 @@
     #
     # FIXME: integrate to grovel-features., maypahps
     printf ' :stack-allocatable-closures' >> $ltf
-    $GNUMAKE -C tools-for-build where-is-mcontext -I src/runtime
-    tools-for-build/where-is-mcontext > src/runtime/ppc-linux-mcontext.h
+    $GNUMAKE -C tools-for-build where-is-mcontext -I ../src/runtime
+    tools-for-build/where-is-mcontext > src/runtime/ppc-linux-mcontext.h || echo "error running where-is-mcontext"
 elif [ "$sbcl_arch" = "ppc" -a "$sbcl_os" = "darwin" ]; then
     printf ' :stack-allocatable-closures' >> $ltf
     # We provide a dlopen shim, so a little lie won't hurt
