--- sbcl-0.9.4/install.sh.lib64	2005-05-28 04:00:34.000000000 -0500
+++ sbcl-0.9.4/install.sh	2005-09-12 14:00:27.310267719 -0500
@@ -13,10 +13,11 @@
 MAN_DIR=${MAN_DIR-$INSTALL_ROOT/share/man}
 INFO_DIR=${INFO_DIR-$INSTALL_ROOT/share/info}
 DOC_DIR=${DOC_DIR-$INSTALL_ROOT/share/doc/sbcl}
+LIB_DIR=${LIB_DIR-$INSTALL_ROOT/lib}
 
 # Does the environment look sane?
 SBCL_SOURCE=`pwd`
-if [ -n "$SBCL_HOME" -a "$INSTALL_ROOT/lib/sbcl" != "$SBCL_HOME" ];then
+if [ -n "$SBCL_HOME" -a "$LIB_DIR/sbcl" != "$SBCL_HOME" ];then
    echo SBCL_HOME environment variable is set, and conflicts with INSTALL_ROOT.
    echo Aborting installation.  Unset one or reset the other, then try again
    echo INSTALL_ROOT="$INSTALL_ROOT"
@@ -37,10 +38,10 @@
     exit 1
 fi
 
-SBCL_HOME=$INSTALL_ROOT/lib/sbcl
+SBCL_HOME=$LIB_DIR/sbcl
 export SBCL_HOME INSTALL_ROOT
 ensure_dirs $BUILD_ROOT$INSTALL_ROOT $BUILD_ROOT$INSTALL_ROOT/bin \
-    $BUILD_ROOT$INSTALL_ROOT/lib  \
+    $BUILD_ROOT$LIB_DIR  \
     $BUILD_ROOT$MAN_DIR $BUILD_ROOT$MAN_DIR/man1 \
     $BUILD_ROOT$INFO_DIR $BUILD_ROOT$DOC_DIR \
     $BUILD_ROOT$DOC_DIR/html \
@@ -76,7 +77,7 @@
 echo
 echo "SBCL has been installed:"
 echo " binary $BUILD_ROOT$INSTALL_ROOT/bin/sbcl"
-echo " core and contribs in $BUILD_ROOT$INSTALL_ROOT/lib/sbcl/"
+echo " core and contribs in $BUILD_ROOT$LIB_DIR/sbcl"
 
 # Installing manual & misc bits of documentation
 #
