diff -up sbcl-1.0.25/make-target-1.sh.default-sbcl-home sbcl-1.0.25/make-target-1.sh
--- sbcl-1.0.25/make-target-1.sh.default-sbcl-home	2006-04-27 11:32:42.000000000 -0500
+++ sbcl-1.0.25/make-target-1.sh	2009-02-03 08:18:10.000000000 -0600
@@ -20,6 +20,29 @@ LANG=C
 LC_ALL=C
 export LANG LC_ALL
 
+# Allow the definition of INSTALL_ROOT and/or SBCL_HOME to correctly
+# set the hard-coded SBCL_HOME macro in src/runtime/runtime.c
+
+DEFAULT_SBCL_HOME=${DEFAULT_SBCL_HOME:-/usr/local/lib/sbcl/}
+
+# Strip off any trailing / on the name; we'll add this later but don't
+# need two
+DEFAULT_SBCL_HOME=${DEFAULT_SBCL_HOME%/}
+
+export DEFAULT_SBCL_HOME
+
+file=src/runtime/sbcl-home.h
+echo "/* This is a machine-generated file.            */" > $file
+echo "/* Please do not edit it by hand.               */" >> $file
+echo "/* Change the default SBCL_HOME by setting the  */" >> $file
+echo "/* DEFAULT_SBCL_HOME environment variable prior */" >> $file
+echo "/* to building. See make-target-1.sh for more   */" >> $file
+echo "/* information.                                 */" >> $file
+echo "#ifndef SBCL_HOME" >> $file
+echo "#define SBCL_HOME \"$DEFAULT_SBCL_HOME/\"" >> $file
+echo "#endif" >> $file
+
+
 # Build the runtime system and symbol table (.nm) file.
 #
 # (This C build has to come after the first genesis in order to get
diff -up sbcl-1.0.25/src/runtime/runtime.c.default-sbcl-home sbcl-1.0.25/src/runtime/runtime.c
--- sbcl-1.0.25/src/runtime/runtime.c.default-sbcl-home	2009-01-25 18:56:09.000000000 -0600
+++ sbcl-1.0.25/src/runtime/runtime.c	2009-02-03 08:19:24.000000000 -0600
@@ -66,9 +66,7 @@
 #include "interr.h"
 #endif
 
-#ifndef SBCL_HOME
-#define SBCL_HOME "/usr/local/lib/sbcl/"
-#endif
+#include "sbcl-home.h"
 
 #ifdef LISP_FEATURE_HPUX
 extern void *return_from_lisp_stub;
