--- sbcl-0.9.5/tools-for-build/where-is-mcontext.c.stdlib_h	2005-07-14 15:24:19.000000000 -0500
+++ sbcl-0.9.5/tools-for-build/where-is-mcontext.c	2005-09-29 14:27:25.840464016 -0500
@@ -20,6 +20,7 @@
 
 #include <stdio.h>
 #include <stddef.h>
+#include <stdlib.h>
 #include <sys/ucontext.h>
 
 int main (int argc, char *argv[]) {
@@ -45,5 +46,7 @@
     }
     printf("\
 #endif /* PPC_LINUX_MCONTEXT_H */\n");
+
+    exit(0);
 }
 
